### DISCUSSION TITLE
Android back button and app history

### DISCUSSION THREAD

**Discussion by NadaComa** [beeware/toga#2644] [Q&A]
Hi all,

I have a button which shows a different box with widgets when pressed. E.g.
`button = toga.Button("Button",on_press=self.switch)
def switch(self, widget):
  self.main_window.content=self.box2
`

Now pressing  the back button closes the app. 

How can I make it to show box1 again. Do I have to catch the "on_exit" event and keep my internal history to  prevent closing the app?

**Comment by freakboy3742** [comment]
Toga doesn't currently have any way to bind to the Android-level "back" button. 

I don't use Android as my daily-driver, so I don't have complete day-to-day experience - but as I understand it, *every* app will exit if you press the back button while on the home screen - that's what the back button is *for*. It's a context sensitive in-app navigation aid. If you're deep in a navigation tree, it takes you back to the previous page of content; if you're at the top level of the app, it takes you "back" to the home page. 

If my understanding is incomplete or incorrect, I'm willing to be corrected; but I know it's *not* just an "exit" button, so attaching handling to on_exit wouldn't be appropriate; In general, Toga isn't likely to provide overrides for functionality that reflects core UX motifs from the operating system.

**Reply by mhsmith** [reply]
Previous discussion: https://github.com/beeware/toga/issues/1623#issuecomment-1961305258

**Comment by NadaComa** [comment]
I see, so without having any clou how the underlying android framework works:
Any easy way to hook into [the back button itself](https://developer.android.com/reference/android/app/Activity#onBackPressed())?
I doubt it but before trying to come up with a different UI approach, I thought I ask :-)

**Reply by mhsmith** [reply]
I think it would require modifying both Toga and the Briefcase Android template.
